Detectors/MUON/MCH/Calibration/src/MCHChannelCalibratorSpec.h

```diff
@@ -84,7 +84,11 @@ class MCHChannelCalibDevice : public o2::framework::Task
   //_________________________________________________________________________________________________
   void run(o2::framework::ProcessingContext& pc) final
   {
-    auto tfcounter = o2::header::get<o2::framework::DataProcessingHeader*>(pc.inputs().get("input").header)->startTime; // is this the timestamp of the current TF?
+    const o2::framework::DataProcessingHeader* header = o2::header::get<o2::framework::DataProcessingHeader*>(pc.inputs().get("input").header);
+    if (!header) {
+      return;
+    }
+    auto tfcounter = header->startTime; // is this the timestamp of the current TF?
```

Comment:
can we get an answer to this question and drop the comment here ?

Reply:
We have actually 2 overlapping counters: the one you use from the DataProcessingHeader and `DataHeader::tfCounter``.
I assume the DataProcessingHeader will disappear soon, so would suggest the 2nd one, but better @ktf confirms.

Reply:
Yes, you probably want to use `DataHeader::tfCounter`, however, as discussed, this is just a counter.

Reply:
Also in TOF I used this as a proxy for the start time of the TF. Is there a better way to retrieve it? 
See:

https://github.com/AliceO2Group/AliceO2/blob/be56bfe25603983275b909970f2b4128e04eef57/Detectors/TOF/calibration/testWorkflow/TOFChannelCalibratorSpec.h#L72

Here I would really need the start time. Should I use DataProcessingHeader->startTime? I am not sure, also considering that @shahor02 mentioned that it will disappear. 

Then I use it as a counter here:

https://github.com/AliceO2Group/AliceO2/blob/be56bfe25603983275b909970f2b4128e04eef57/Detectors/TOF/calibration/testWorkflow/LHCClockCalibratorSpec.h#L51
https://github.com/AliceO2Group/AliceO2/blob/be56bfe25603983275b909970f2b4128e04eef57/Detectors/TOF/calibration/testWorkflow/TOFCalibCollectorSpec.h#L53

Should we simply replace this call with

`auto tfcounter = o2::header::get<o2::framework::DataHeader*>(pc.inputs().get("input").header)->tfCounter`

? Note also that everybody is "copying" from this (e.g. Detectors/EMCAL/calibration/testWorkflow/EMCALChannelCalibratorSpec.h), so if it should be changed, we need to do it everywhere.

Thanks for any clarification...

Reply:
@chiarazampolli strictly speaking, this startTime is the time of the STF building on the FLP machine, which is not the same as the "CTP" time corresponding to the 1st orbit of the TF (which should eventually go to the CCDB and be used for queries).
The real time stamp should be calculated as "ctp_time_of_orbit0" + BCcount_of_TF_1st_orbit, but this is not available at the moment. So, for time-like parameter one can use ``startTime`` until the real one will be available. 
Then can be fixed everywhere in a centralized way.

Reply:
Hi @shahor02 , 

Ok, but then I would not change it anywhere, to not make mistakes. Even if for the TimeSlotCalibration a counter can do the same job as a real timestamp, also there we assumed to use eventually use timestamps (note that now that we simulate the input, this startTime is indeed working as a counter, nothing more). What I have doubts about is the fact that you said that 

> DataProcessingHeader will disappear soon

One thing we can do is to change that comment to something like "should be replaced with real start time of the TF".

Chiara

Reply:
There were plans to get rid of the DataProcessingHeader completely... DD will not be able to put in any data header the "real start time" of the TF, instead, it can be calculated based on the TF 1st orbit. I think the best would be to calculate it in the workflow, either in the DPL or by dedicated device and feed it as a message.